### PR TITLE
Cpp merge (#10)

### PR DESCRIPTION
**Pull request by petrovicboban**

### CODE REVIEW

File: .github/workflows/python.yaml [hunk 1]
```diff
@@ -61,10 +61,9 @@ jobs:
         uses: pypa/cibuildwheel@v2.11.2
         env:
           CIBW_ARCHS: x86_64
+          CIBW_MANYLINUX_X86_64_IMAGE: ghcr.io/petrovicboban/rforestry/python_build_env:docker
```
Comment (petrovicboban) [inline .github/workflows/python.yaml:64]: Please check if "packages" are enabled in your github account.
we'll publish docker image to the github registry, so we'll have to change account name here

**Comment by edwardwliu** [comment]
Should we restore `#include <armadillo`  in the place of `#include <RcppArmadilllo` so this merge isn't breaking? We can then resolve any merge conflict of `<armadillo` / `<RcppArmadillo>` in a separate PR.

**Comment by petrovicboban** [comment]
Here is the PR of my feature branch to my master branch with the change above https://github.com/petrovicboban/Rforestry/pull/11
It seems it resolved the issue with `RcppArmadillo` but other issue appeared.

**Comment by theo-s** [comment]
Comment to check notifications

**Comment by edwardwliu** [comment]
Based on our meeting today, let's move forward with the following:

1. Boban, can you create a separate PR for the custom Docker image change? Theo just granted you full access to the repository. Let me know if you need additional permissions in order to update settings! We should also test the Python package is building well at this step. 

3. Originally missed that the RcppArmadillo change was made in a [PR](https://github.com/petrovicboban/Rforestry/pull/11) into your fork. Theo will take a look, I think it should be resolvable as a pure C++ non-platform issue. Let's land that change then we can get this PR fully merged.

**Comment by theo-s** [comment]
There may be a few other differences based on functions we have removed from the C++ API recently, for example symmetry [here](https://github.com/petrovicboban/Rforestry/blob/f7023b3b55539642779c952f8980d5e82fd29913/C%2B%2B/src/api.cpp#L175). So I agree with Edward that these are most likely all pure C++ issues and should be pretty straightforward to resolve.

**Comment by theo-s** [comment]
If you want me to take a pass, I can probably resolve these pretty quickly since we recently made these changes in the other repo.

**Comment by edwardwliu** [comment]
That's great @theo-s, since you're already familiar with the API changes, could you update `api.cc`? May have to get those changes (and the swap to armadillo) into Boban's master fork to update this PR.

**Comment by petrovicboban** [comment]
I will remove this PR and create the same one, but from branch in Theo's repo, so it's easier for everyone to make changes.

**Comment by edwardwliu** [comment]
For posterity, moving the discussion and `api.cc` changes to PR https://github.com/theo-s/Rforestry/pull/19.

**Pull request closed without merging**